### ISSUE TITLE
Expose path cleaning

### ISSUE THREAD

**Issue by cppforlife** [charlievieth/fs#1]
@charlievieth what do you think about exposing some kind of function (absPath) to resolve paths so that generic code that works on windows/unix doesnt have to do cleaning? Here is an example PR that includes cleaning functionality:

https://github.com/cloudfoundry/bosh-utils/pull/11

**Comment by charlievieth** [comment]
I do not believe this can be done as the Windows versions of `posix` utilities differ in what paths they accept.  [PR #11](https://github.com/cloudfoundry/bosh-utils/pull/11) works with the tar included with [git-bash](https://git-scm.com/downloads), but breaks the tar included with cygwin and libarchive, which properly handles Windows paths.

That said, I've been debating exposing `absPath`, but I'm not sure how valuable it would be, as outside of the `os` package, which `fs` fully replicates, the most common situation in which `MAX_PATH` is violated is when walking directories and `absPath` does not know if a path will be walked.  To handle this issue I do plan to add implement `filepath.Walk` with support for long paths in the `fs` package.
